src/main/java/org/sonarlint/intellij/tasks/BindingStorageUpdateTask.java

```diff
@@ -149,40 +150,49 @@ public void doRun() {
+    Collection<Project> projectsToUpdate) {
     var failures = new ArrayList<ProjectStorageUpdateFailure>();
 
-    Map<String, String> branchByProjectKey = new HashMap<>();
+    Set<String> allProjectKeysToSync = new HashSet<>();
     for (Project project : projectsToUpdate) {
-      var vcsService = getService(project, VcsService.class);
-      var modules = ModuleManager.getInstance(project).getModules();
-      for (Module module : modules) {
-        var projectKey = getService(module, ModuleBindingManager.class).resolveProjectKey();
-        var serverBranchName = vcsService.getServerBranchName(module);
-        // in theory it's possible that different (key, branch) pairs can exist
-        branchByProjectKey.put(projectKey, serverBranchName);
-      }
+      ProjectBindingManager bindingManager = getService(project, ProjectBindingManager.class);
+      var projectKeysToSync = bindingManager.getUniqueProjectKeys();
+      allProjectKeysToSync.addAll(projectKeysToSync);
     }
 
-    branchByProjectKey.forEach((projectKey, branchName) -> {
+    allProjectKeysToSync.forEach(projectKey -> {
       try {
-        engine.updateProject(connection.getEndpointParams(), connection.getHttpClient(), projectKey, true, branchName, monitor);
+        engine.updateProject(connection.getEndpointParams(), connection.getHttpClient(), projectKey, monitor);
       } catch (Exception e) {
         GlobalLogOutput.get().logError(e.getMessage(), e);
         failures.add(new ProjectStorageUpdateFailure(projectKey, e));
       }
     });
+    engine.sync(connection.getEndpointParams(), connection.getHttpClient(), allProjectKeysToSync, monitor);
+
+    updateAllProjectIssuesForCurrentBranch(engine, connection, monitor, projectsToUpdate);
 
-    engine.sync(connection.getEndpointParams(), connection.getHttpClient(), branchByProjectKey.keySet(), monitor);
     projectsToUpdate.forEach(project -> project.getMessageBus().syncPublisher(ProjectSynchronizationListenerKt.getPROJECT_SYNC_TOPIC()).synchronizationFinished());
 
     return failures;
   }
 
+  private static void updateAllProjectIssuesForCurrentBranch(ConnectedSonarLintEngine engine, ServerConnection connection, TaskProgressMonitor monitor,
+    Collection<Project> projectsToUpdate) {
+    var allProjectAndBranchesToSync = new HashSet<ProjectBindingManager.ProjectKeyAndBranch>();
+    for (Project project : projectsToUpdate) {
+      ProjectBindingManager bindingManager = getService(project, ProjectBindingManager.class);
+      var projectAndBranchesToSync = bindingManager.getUniqueProjectKeysAndBranchesPairs();
+      allProjectAndBranchesToSync.addAll(projectAndBranchesToSync);
+    }
+    allProjectAndBranchesToSync
+      .forEach(pb -> engine.downloadAllServerIssues(connection.getEndpointParams(), connection.getHttpClient(), pb.getProjectKey(), pb.getBranchName(), monitor));
```

Comment:
Why not use `syncServerIssues` here ? You don't want to use the incremental update but reset the storage ?

Reply:
This is bit misleading, but `syncServerIssues` is a noop for older SQ versions and SC. Here we want to refresh issues for all SQ versions.